### ISSUE TITLE
Function header documentation for OS_ObjectIdAllocateNew() is incorrect

### ISSUE THREAD

**Issue by nohlson** [nasa/osal#1242]
Describe the bug
The comments in the function header for OS_ObjectIdAllocateNew() is outdated/incorrect. I see that this was previously reported in #17 but it appears that since then the function signature and implementation has again changed. https://github.com/nasa/osal/blob/88f72d788d33a12e7bf9d9f01c77346438730ffa/src/os/shared/src/osapi-idmap.c#L1169

To Reproduce
N/A. Only in the function comments.

Expected behavior
Description of the function should be with the function prototype as described in #17. In this case the removal of the Inputs and Outputs sections of the function header.

Reporter Info
Nathan Ohlson

**Comment by skliper** [comment]
Agreed, many of function comments in osapi-idmap.c repeat, extend, or even occasionally contradict what's in os-shared-idmap.h.    Should consolidate into os-shared-idmap.h and reference that documentation from osapi-idmap.c.
